core/controller/src/main/scala/whisk/core/loadBalancer/ContainerPoolBalancer.scala

```diff
@@ -345,7 +345,8 @@ object ContainerPoolBalancer extends LoadBalancerProvider {
         .find(_._3 < invokerBusyThreshold)
         .orElse(invokerProgression.find(_._3 < invokerBusyThreshold * 2))
         .orElse(invokerProgression.find(_._3 < invokerBusyThreshold * 3))
-        .orElse(invokerProgression.headOption)
+        .orElse(
+          if (invokerProgression.size == 0) None else invokerProgression.lift(Random.nextInt(invokerProgression.size)))
```

Comment:
The extra check for `size == 0` is redundant I think, since `lift` will automatically bound-check anyway.

It is **crucial** to use `ThreadLocalRandom` vs. `Random` because getting randomness is synchronized and thus will result in thread-contention.

Reply:
`invokerProgression` here,  doesn't have `nonEmpty` method.

Reply:
@markusthoemmes , can you look at this again?